auction create Page Input element Validation logic

FioreFlower opened the pull request:
## PR 타입(하나 이상의 PR 타입을 선택해주세요)
- [x] 기능 추가

<br/>

## 반영 브랜치
jinyeong/29-auction-create-page-business-logic-develop -> dev

<br/>

## 👨🔧 변경사항
<!-- 이 PR에서 어떤점들이 변경되었는지 기술해주세요. 가급적이면 as-is, to-be를 활용해서 작성해주세요.  -->
- **AS-IS**
  - Input Element 입력값의 유효성 검사를 하지 않아 잘못된 값이 넘어올 가능성이 있었음
  - 입력 가능한 데이터 타입과 값 범위가 정해지지 않아 부적절한 값이 입력될 가능성이 있었음

- **TO-BE**
  - 잘못된 값을 입력할 수 없도록 유효성 검사를 실시하고 유효하지 않은 값은 입력을 받지 않도록 함
  - 입력 가능한 데이터 타입과 값의 범위를 제한하여 유효한 값이 입력될 수 있도록 설정
<br />

## 🎨 이미지 첨부
<!-- 이미지 첨부는 자유 입니다. 하실분은 하시고 안하실 분은 안하셔도 됩니다. -->

<br/>

## 👨💻 테스트
<!-- 본 변경사항이 테스트가 되었는지 기술해주세요 --> 
- [x] 테스트 코드
- [x] API 테스트
<br/>

## ➕ 이슈 링크

- [Front-End #29](https://github.com/Fan-Tion/Front-End/issues/29)

DoHyeongKwonn committed:
feat: 경매 내역, 예치금 입출금 내역 뷰 생성

-경매 내역 페이지와 예치금 입출금 페이지의 뷰를 생성했습니다.
-Route는 테스트를 위해 임시로 작성해두었습니다.

DoHyeongKwonn committed:
Merge branch 'main' into dohyeong/history-page-create

Min-byeonghyun committed:
HOTFIX : 파싱에러 } 빠진거 수정

DoHyeongKwonn committed:
feat : #12 경매 내역, 예치금 입출금 내역 및 Mock API 작성

-주요 로직 구현
-Moci API 임의작성 이후 수정 필요
이슈 #12

Min-byeonghyun committed:
feat : #27 : 패스워드변경 페이지 뷰 , 비밀번호찾기 핸들러 작성, 비밀번호변경 라우트설정 완료

DoHyeongKwonn committed:
fix : #12 MSW db.ts api 명세서에 맞게 수정

db에 작성했던 목데이터를 api명세서에 맞게 수정했습니다.
이슈 #12

DoHyeongKwonn committed:
Merge branch 'main' into dohyeong/history-page-create

Min-byeonghyun committed:
fix : 비밀번호 변경 핸들러 ,로직 추가 후 테스트 완료

Min-byeonghyun committed:
fix : #27 비밀번호 변경 핸들러, 로직 추가 후 테스트 완료

FioreFlower committed:
feat :  form input 유효성 검사 추가

- 각 input 요소의 유효성을 검사할 수 있는 로직을 추가함
- 요구사항에 충족할 수 있도록 입력 가능한 데이터 타입 정의
- 입력 가능한 범위를 제한하는 로직 추가

F/U #29

Min-byeonghyun committed:
프로필페이지 내정보 뿌려주기 완료

FioreFlower committed:
style : 코드 스타일 변경

- 기능에 영향을 주지 않는 코드 변경

F/U #29

FioreFlower committed:
fix : 즉시 구매가가 경매 시작가보다 작거나 같을 수 없도록 수정

- 즉시구매가 유효성 검사 누락되어 추가

F/U #29

DoHyeongKwonn commented on src/components/AuctionCreatePageComponents/index.tsx:
```diff
@@ -30,18 +30,120 @@ const Button = styled.button`
+      [name]: value.trim() !== '',
+    });
+  }
+
```
유효성 검사를 어떤식으로 하는건지 궁금했는데 코드를 보니 이해가 잘 됐습니다 고생하셨습니다.

DoHyeongKwonn committed:
fix : #12 axios를 axiosInstance로 수정

axios로 보내던 요청을 axiosInstance로 사용하도록 수정했습니다
이슈 #12

DoHyeongKwonn committed:
Merge pull request #28 from Fan-Tion/dohyeong/history-page-create

경매, 예치금 내역 페이지 생성

Min-byeonghyun committed:
스타일수정

Min-byeonghyun committed:
dev브랜치

Min-byeonghyun committed:
byeonghyun/myinfo-create ->  dev 병합

FioreFlower committed:
fix : 이미지 컴포넌트 버튼 타입 문제 해결

- 폼 입력후 엔터키 입력시 이미지가 삭제되는 이슈 해결
- 즉시 구매가 유효성 검사 수정

F/U #29

Min-byeonghyun committed:
Merge pull request #33 from Fan-Tion/dev

byeonghyun/myinfo-create -> dev 병합후 테스트완료

Min-byeonghyun committed:
GlobalInput, GlobalButton 스타일 설정

Min-byeonghyun committed:
글로벌 스타일 수정

Min-byeonghyun committed:
수정

Min-byeonghyun committed:
Merge pull request #34 from Fan-Tion/byeonghyun/global-input-button-create

Byeonghyun/global input button create

FioreFlower committed:
fix : 즉시 구매가, 경매 시작가가 0일 경우 빈 문자열로 표시되도록 수정

- 폼데이터 포멧 수정하여 값이 0일 때는 빈 문자열로 표시되도록 처리

F/U #29

FioreFlower committed:
Merge branch 'jinyeong/29-auction-create-page-business-logic-develop' into jinyeong/dev

* jinyeong/29-auction-create-page-business-logic-develop:
  fix : 즉시 구매가, 경매 시작가가 0일 경우 빈 문자열로 표시되도록 수정
  fix : 이미지 컴포넌트 버튼 타입 문제 해결
  fix : 즉시 구매가가 경매 시작가보다 작거나 같을 수 없도록 수정
  style : 코드 스타일 변경
  feat :  form input 유효성 검사 추가

FioreFlower committed:
feat : 모달 추가, 디자인 수정

- 이미지 업로드 컴포넌트를 모달에서 사용하도록 수정
- 색상 및 버튼 디자인 통일
- DateRange 함수 훅으로 변경
- 범용 모달 컴포넌트 생성

F/U #29

FioreFlower committed:
feat : 입력 스타일 조정, 모달 위치 조정 버튼 함수 추가

- 작성 취소 버튼으로 홈 화면으로 돌아갈 수 있음
- 입력 요소 디자인 요소 일부 수정
- 모달 위치 수정

F/U #29

FioreFlower committed:
feat : 폼데이터 업데이트, 코드 분리

- 이미지 데이터를 폼데이터에 담을 수 있도록 업데이트
- 선택 가능한 날짜를 제한하는 메서드 코드를 옮김

F/U #29

FioreFlower committed:
fix : api 명세서에 맞게 변수명 과 데이터 구조 수정

- 드래그 가능한 이미지 컴포넌트에 key가 고유할 수 있도록 수정

F/U #29

FioreFlower committed:
style : 사용되지 않는 변수앞에 _ 삽입

- file 변수를 직접 사용하진 않으므로 lint 규칙에 따라 처리

F/U #29

FioreFlower committed:
feat : 서버에 데이터 전달하는 코드 작성

- submit 실행시 버튼 비활성화 하고 완료시 다시 활성화함
- dateRange 함수 훅으로 사용하므로 이름 재설정
- axios 설정 이용해서 데이터 연결 진행

F/U #29
